packages/valory/skills/oracle_deployment_abci/rounds.py

```diff
@@ -201,4 +228,4 @@ class OracleDeploymentAbciApp(AbciApp[Event]):
         Event.VALIDATE_TIMEOUT: 30.0,
         Event.DEPLOY_TIMEOUT: 30.0,
     }
-    cross_period_persisted_keys = ["oracle_contract_address"]
+    cross_period_persisted_keys = ["safe_contract_address", "oracle_contract_address"]
```

Comment:
We could use `get_name` here as well. Also I think we can add the `safe_contract_address` to the cross period keys of registration abci. I don't think we should rely on the users to do this.

Reply:
Agreed! Though this is not available here yet, it will be done when bumping to open-autonomy `v0.6.0`.

Regarding the `safe_contract_address`, I also think this is a nice idea! One thing to keep in mind is that some users or apps might not need it (such as the APY oracle) and therefore it will be in the cross-period keys without a reason, but I think it is a small trade-off.